src/libs/Violations/ViolationsUtils.ts

```diff
@@ -119,7 +120,12 @@ const ViolationsUtils = {
         policyTagList: PolicyTagList,
         policyRequiresCategories: boolean,
         policyCategories: PolicyCategories,
-    ): OnyxUpdate {
+    ): OnyxUpdate | null {
+        const isPartialTransaction = TransactionUtils.isPartialMerchant(TransactionUtils.getMerchant(updatedTransaction)) && TransactionUtils.isAmountMissing(updatedTransaction);
+        if (isPartialTransaction) {
+            return null;
```

Comment:
Can we move this logic under `let newTransactionViolations = [...transactionViolations];` and return the following if the transaction is partial? I think if we do that we don't need any of the other changes and the logic is simpler
```js
        return {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${updatedTransaction.transactionID}`,
            value: newTransactionViolations,
        };
```

Reply:
That's nice. Updated. Btw, do you think we can just return an empty array instead? Is it possible a transaction already has a violation while it's still a partial transaction?

Reply:
It shouldn't, but the effect should be the same, right?

Reply:
If it's not possible then yes, the effect is the same.